pl_examples/hydra_examples/pl_hydra/trainer_conf.py

```diff
@@ -243,3 +243,36 @@ class WandbConf:
     node=ObjectConf(cls="pytorch_lightning.loggers.wandb.WandbLogger", params=WandbConf()),
 )
 
+
+@dataclass
+class TrainerIn(LightningTrainerConf):
+    logger: Optional[Any] = None
+    profiler: Optional[Any] = None
+    checkpoint_callback: Optional[Any] = None
+    early_stop_callback: Optional[Any] = None
+
+
+def init_trainer(cfg):
+    logger = None
+    profiler = None
+    early_stopping = None
+    checkpoint = None
+    if "logger" in cfg.pl:
+        logger = hydra.utils.instantiate(cfg.pl.logger)
```

Comment:
Anything else needs access to these objects, or is it just the trainer?

Maybe a big cleaner:
```python
def conf_init(cfg, key):
  return hydra.utils.instantiate(getattr(cfg, key)) if hasattr(cfg,  key) else None

profiler = conf_init(cfg.pl, "profiler"))
early_stopping = conf_init(cfg.pl, "early_stopping "))
checkpoint = conf_init(cfg.pl, "checkpoint ofiler"))
```

Reply:
Good idea!